lib/systems/default.nix

```diff
@@ -79,13 +77,18 @@ rec {
       };
       isStatic = final.isWasm || final.isRedox;
 
-      kernelArch =
```

Comment:
I think this should be separate from platform / linux-kernel. This should be a direct mapping from cpu name to linux kernel name. We already have qemuArch and will soon have a "darwinArch" in https://github.com/NixOS/nixpkgs/pull/105026. This shouldn't be configurable like platform / kernel-arch is.